src/openqaoa-core/problems/portfoliooptimization.py

```diff
@@ -0,0 +1,223 @@
+from .converters import FromDocplex2IsingModel
+from .qubo import QUBO
+
+
+class PortfolioOptimization(Problem):
+    """
+    Creates an instance of the portfolio optimization problem.
+    https://en.wikipedia.org/wiki/Portfolio_optimization
+
+    Parameters
+    ----------
+    mu: List[int]
+        The expected return
+    sigma: List[List[int]]
+        The covariance matrix.
+    risk_factor: float
+        desired risk factor exposures
+    budget: int
+        maximum number of assets to buy
+    penalty: float
+        Penalty for the maximum budget.
+
+    Returns
+    -------
+        An instance of the portfolio optimization problem.
+    """
+
+    __name__ = "portfolio_optimization"
+
+    def __init__(self, mu, sigma, risk_factor, budget, penalty):
+        # Check whether the input is valid. Number of values should match the number of weights.
+        for s_i in sigma:
+            if len(mu) != len(s_i):
+                raise ValueError("Number of assets does not match sigma")
+        self.mu = mu
+        self.sigma = sigma
+        self.risk_factor = risk_factor
+        self.penalty = penalty
+        self.num_assets = len(mu)
+        self.budget = budget
+
+    @staticmethod
+    def random_instance(**kwargs):
+        """
+        Creates a random instance of the Knapsack problem.
+
+        Parameters
+        ----------
+        num_assets: int
+            The number of assets.
+        mu_bounds: tuple
+            bounds of the expected return
+        sigma_bounds: tuple
+            bound of the covariance matrix
+        risk_factor: float
+            desired risk factor exposures
+        seed: int
+            Seed for the random number selection.
+
+        Returns
+        -------
+            A random instance of the portfolio optimization problem.
+        """
+        num_assets = kwargs.get("num_assets", 8)
+        seed = kwargs.get("seed", 1234)
+        mu_bounds = kwargs.get("mu_bounds", (-1, 1))
+        sigma_bounds = kwargs.get("sigma_bounds", (-0.1, 0.1))
+        budget = kwargs.get("budget", 5)
+        risk_factor = kwargs.get("risk_factor", 0.1)
+        penalty = kwargs.get("penalty", 2)
+
+        np.random.seed(seed)
+
+        mu = [
+            (mu_bounds[1] - mu_bounds[0]) * np.random.rand() + mu_bounds[0]
+            for _ in range(num_assets)
+        ]
+        sigma = [[0 for i in range(num_assets)] for j in range(num_assets)]
+        for i in range(num_assets):
+            for j in range(num_assets):
+                sigma[i][j] = (
+                    sigma_bounds[1] - sigma_bounds[0]
+                ) * np.random.rand() + sigma_bounds[0]
+
+        return PortfolioOptimization(mu, sigma, risk_factor, budget, penalty)
+
+    @property
+    def docplex_model(self):
+        mdl = Model("portfolio_optimization")
+
+        num_assets = self.num_assets
+
+        # Start the docplex model with Model("name of the model")
+        mdl = Model("Portfolio Optimization")
```

Comment:
@alejomonbar, the `mdl` is defined twice. I believe that's a typo. Can you remove one?

Reply:
done!